client/state/notification-settings/actions.js

```diff
@@ -24,3 +39,90 @@ export const updateNotificationSettings = settings => ( {
 	type: NOTIFICATION_SETTINGS_UPDATE,
 	settings,
 } );
+
+export const toggle = ( source, stream, setting ) => dispatch => {
+	dispatch( {
+		type: NOTIFICATION_SETTINGS_TOGGLE_SETTING,
+		source,
+		stream,
+		setting,
+	} );
+};
+
+export const toggleWPcomEmailSetting = setting => toggle( 'wpcom', 'email', setting );
+
+export const fetchSettings = () => dispatch => {
+	dispatch( { type: NOTIFICATION_SETTINGS_FETCH } );
+
+	wpcom
+		.undocumented()
+		.me()
+		.getNotificationSettings( ( error, data ) => {
```

Comment:
We usually use the promise version in new code, and it would make this handler look a bit better, too.

Reply:
Agreed! I figured I'd just get it copied through but it is technically a new file - I'll update soon :)